src/components/ThoughtAnnotation.tsx

```diff
@@ -279,13 +284,78 @@ const ThoughtAnnotationContainer = React.memo(
       setCalculateContexts(true)
     }, [])
 
+    const positionAnnotation = useCallback(() => {
```

Comment:
This could use a header comment summarizing the purpose of this function.

It may also be worth it to pull `positionAnnotation` and the `useEffect` out into a hook in order to hide the implementation details from the rest of the component.